File: VisualIpcvEditor/src/com/visualipcv/core/ProcessorProperty.java
package com.visualipcv.core;

public class ProcessorProperty {
    private String name;
    private DataType type;
    private boolean showControl = true;
    private boolean showConnector = true;

    public ProcessorProperty(String name, DataType type) {
        this.name = name;
        this.type = type;
    }

    public ProcessorProperty(String name, DataType type, boolean showControl, boolean showConnector) {
        this.name = name;
        this.type = type;
        this.showControl = showControl;
        this.showConnector = showConnector;
    }

    public String getName() {
        return name;
    }

    public DataType getType() {
        return type;
    }

    public boolean showControl() {
        return showControl;
    }

    public boolean showConnector() {
        return showConnector;
    }
}

File: Native/VisualIPCV/ProcessorProperty.cpp
//
//  ProcessorProperty.cpp
//  VisualIPCV
//
//  Created by Артём Новиков on 15.02.2020.
//

#include "ProcessorProperty.hpp"

ProcessorProperty::ProcessorProperty() {
    name = "";
    type = BaseDataType();
}

ProcessorProperty::ProcessorProperty(const ProcessorProperty & object) {
    this->name = object.name;
    this->type = BaseDataType(object.type);
}

ProcessorProperty::ProcessorProperty(std::string name, BaseDataType type) {
    this->name = name;
    this->type = BaseDataType(type);
}
